<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MorphDialog</class>
 <widget class="QDialog" name="MorphDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>576</width>
    <height>258</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBoxMorphologicalOperations">
       <property name="title">
        <string>Morphological Operations</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QLabel" name="labelMorphologicalOperation">
              <property name="text">
               <string>Operation: </string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="comboBoxMorphologicalOperation">
              <item>
               <property name="text">
                <string>Dilation</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Erosion</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Opening</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Closing</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Morphological Gradient</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Top Hat</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Black Hat</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="labelMorphologicalIterationCount">
              <property name="text">
               <string>Iteration Count: </string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QSpinBox" name="spinBoxMorphologicalIterationCount">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QLabel" name="labelMorphologicalKernelSize">
              <property name="text">
               <string>Kernel Size: </string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinBoxMorphologicalKernelSize">
              <property name="suffix">
               <string/>
              </property>
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>21</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="labelMorphologicalKernelType">
              <property name="text">
               <string>Kernel Type: </string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QComboBox" name="comboBoxMorphologicalKernelType">
              <item>
               <property name="text">
                <string>Rectangle</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Cross</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Ellipse</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QLabel" name="labelMorphologicalKernelAnchorX">
              <property name="text">
               <string>Kernel Anchor X: </string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinBoxMorphologicalKernelAnchorX">
              <property name="minimum">
               <number>-1</number>
              </property>
              <property name="maximum">
               <number>21</number>
              </property>
              <property name="value">
               <number>-1</number>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="labelMorphologicalKernelAnchorY">
              <property name="text">
               <string>Kernel Anchor Y: </string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QSpinBox" name="spinBoxMorphologicalKernelAnchorY">
              <property name="minimum">
               <number>-1</number>
              </property>
              <property name="maximum">
               <number>21</number>
              </property>
              <property name="value">
               <number>-1</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="3" column="0">
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="QLabel" name="labelMorphologicalPaddingMethod">
              <property name="text">
               <string>Image Padding Method: </string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="comboBoxMorphologicalPaddingMethod">
              <item>
               <property name="text">
                <string>Border Replicate - aaaaaa|abcdefgh|hhhhhhh</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Border Reflect - fedcba|abcdefgh|hgfedcb</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Border Reflect 101 - gfedcb|abcdefgh|gfedcba</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Border Wrap - cdefgh|abcdefgh|abcdefg</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>MorphDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>MorphDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
